### PR TITLE
Add Groovy and Kotlin code to RabbitMQ guide

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy [hunk 1]
```diff
@@ -126,12 +126,17 @@ class GuideProjectGenerator implements Closeable {
             }
 
             for (App app: metadata.apps) {
-                List<String> appFeatures = app.features
+                List<String> appFeatures = [] + app.features
```
Comment (sdelamo) [inline buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy:129]: why is this necessary?

File: guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsController.groovy [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Controller('/analytics')
+class AnalyticsController {
+
+    private final AnalyticsService analyticsService
+
+    AnalyticsController(AnalyticsService analyticsService) {
+        this.analyticsService = analyticsService
+    }
+
+    @Get('/') // <1>
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsController.groovy:17]: This could be `@Get // <1>`

File: guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsController.groovy [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    List<BookAnalytics> listAnalytics() {
+        return analyticsService.listAnalytics()
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsController.groovy:19]: we could remove the optional `return`

**Review by ilopmar — APPROVED** [review]

File: guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsService.groovy [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+@CompileStatic
+@Singleton
+class AnalyticsService {
+
+    private final Map<Book, Long> bookAnalytics = new ConcurrentHashMap<>() // <1>
+
+    void updateBookAnalytics(Book book) { // <2>
+        bookAnalytics.compute(book, (k, v) -> {
+            if (v == null) {
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsService.groovy:16]: Maybe rewrite this: 

```groovy
            if (v == null) {
                 return 1L
             } else {
                 return v + 1
             }            
```


as: 

`v == null ? 1L : v + 1`

File: guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsService.groovy [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                return 1L
+            } else {
+                return v + 1
+            }
+        })
+    }
+
+    List<BookAnalytics> listAnalytics() { // <3>
+        return bookAnalytics.collect { e -> new BookAnalytics(e.key.isbn, e.value) }
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/main/groovy/example/micronaut/AnalyticsService.groovy:25]: Maybe, remove `return` keyword.

File: guides/micronaut-rabbitmq/analytics/groovy/src/main/resources/application.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+micronaut:
+  application:
+    name: analytics
+---
+#tag::port[]
+micronaut:
+  server:
+    port: 8081 # <1>
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/main/resources/application.yml:8]: I think this should use such a strategy: https://github.com/micronaut-projects/micronaut-guides/pull/353

File: guides/micronaut-rabbitmq/analytics/groovy/src/test/groovy/example/micronaut/AnalyticsServiceSpec.groovy [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        3 == findBookAnalytics(b1, analytics).count
+        1 == findBookAnalytics(b2, analytics).count
+    }
+
+    private BookAnalytics findBookAnalytics(Book b, List<BookAnalytics> analytics) {
+        BookAnalytics bookAnalytics = analytics.find { it.bookIsbn == b.isbn }
+        if (!bookAnalytics) {
+            throw new RuntimeException('Book not found')
+        }
+        return bookAnalytics
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/groovy/src/test/groovy/example/micronaut/AnalyticsServiceSpec.groovy:38]: remove `return`

File: guides/micronaut-rabbitmq/analytics/kotlin/src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+micronaut:
+  application:
+    name: analytics
+---
+#tag::port[]
+micronaut:
+  server:
+    port: 8081 # <1>
```
Comment (sdelamo) [inline guides/micronaut-rabbitmq/analytics/kotlin/src/main/resources/application.yml:8]: I think we should do something like: https://github.com/micronaut-projects/micronaut-guides/pull/353

**Review by sdelamo — CHANGES_REQUESTED** [review]
I know @ilopmar merged this PR but I think we should do at least the port changes using default environment. see: https://github.com/micronaut-projects/micronaut-guides/pull/353

File: buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy (hunk 1 above)
Comment (burtbeckwith) [reply buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy:129]: I saw this when removing the mockito feature for Spock - if Groovy is processed before Kotlin or Java, the removal affects those too. This wasn't noticed for GraalVM because the build still compiles and tests pass, but it was causing the Java/Kotlin apps to fail because they didn't have the mockito dependency. Making a copy for each loop iteration makes the change local to that loop iteration and doesn't affect Java or Kotlin, only the current Groovy app
